Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/LeaderBoard.hs

```diff
@@ -68,17 +70,20 @@ getDailyDriverLeaderBoard (personId, merchantId, merchantOpCityId) day = do
   driversWithScoresMap :: [(Text, Double)] <- concat <$> Redis.withNonCriticalRedis (Redis.get $ RideEndInt.makeCachedDailyDriverLeaderBoardKey merchantId day)
   let driverIds = map (Id . fst) driversWithScoresMap
   driverNamesMap :: HM.Map Text (Maybe Text) <- HM.fromList . map (\driver -> (driver.id.getId, getPersonFullName driver)) <$> QPerson.getDriversByIdIn driverIds
+  driverGender <- HM.fromList . map (\driver -> (driver.id.getId, Just driver.gender)) <$> QPerson.getDriversByIdIn driverIds
```

Comment:
we are doing this query `QPerson.getDriversByIdIn driverIds` in the line above as well, just do that once and use it for both `driverNamesMap` and `driverGender`